vllm/compilation/backends.py

```diff
@@ -752,6 +754,27 @@ def __call__(
             self.split_gm, submod_names_to_compile, self.vllm_config, self
         ).run(*example_inputs)
 
+        from torch._guards import detect_fake_mode
+
+        fake_mode = detect_fake_mode()
+
+        if (
```

Comment:
Why do we only do this with `evaluate_guards` on?

Reply:
Oh is it because otherwise we drop all guards anyway?

Reply:
yes